src/vehicle_selector.h

```diff
@@ -12,9 +12,9 @@ struct tripoint;
 class vehicle_cursor : public visitable<vehicle_cursor>
 {
     public:
-        vehicle_cursor( vehicle &veh, int part ) : veh( veh ), part( part ) {};
+        vehicle_cursor( vehicle &veh, std::ptrdiff_t part ) : veh( veh ), part( part ) {};
         vehicle &veh;
-        int part;
+        std::ptrdiff_t part;
```

Comment:
What MSVC warning does this change fix?

Reply:
Conversion from 64-bit int to 32-bit int (loss of data). I.e. size_t to int.

Reply:
ah, wouldn't it be better to change at the point of conversion? Sounds like passing in the 64-bit type is the error, not the storage type being too small. Most of the instances for use I saw were passing in an int as expected.

Reply:
The motivation for this possible resolution (originally `size_t`) was that `vpart_position` uses `size_t` for `part_index_` rather than `int`. So, technically this should be fine as a `size_t`, granted it would also be fine as an `unsigned` or even `unsigned short` given that there are unlikely to be 64k parts in a vehicle.

Reply:
Why `std::ptrdiff_t`? I guess part cannot be negative, so you better use `size_t`.

Reply:
I used a `size_t` and someone said to use `ptrdiff_t`... So, you (collective reviewers) should decide on one and let me know.

Reply:
I previously commented that changing a signed integer to unsigned could cause problems, that's why @ahmedcharles changed it to `std::ptrdiff_t`. `size_t` will of course be better if `part` is not involved in signed operation anywhere.